### DISCUSSION TITLE
Can I fetch data from blockchain api in the guest code?

### DISCUSSION THREAD

**Discussion by silicology1** [risc0/risc0#2287] [General]
What if I feed inputs from blockchain data? Can I fetch data from blockchain api (say [subxt](https://github.com/paritytech/subxt/blob/master/subxt/examples/light_client_basic.rs) in polkadot using lightclient) in my guest code and use it in the code, and then submit the proofs to blockchain. Will their be any security issue??

**Comment by nuke-web3** [comment]
We are deprecating discussions on Github, and encouraging everyone to [join us on discord](https://discord.com/invite/risczero) where we can chat and provide support.

### [Join our Community Discord for dev chit-chat & our support forum](https://discord.com/invite/risczero)

For general questions and problems you are facing with RISC Zero, please **search** our `🙋｜dev-questions-support-forum` first for solutions, by keyword and tags. If you discover no solution, please then ask and questions in our community! We highly encourage everyone also share their understanding by answering questions for others!

**Reply by nuke-web3** [reply]
@amiyatulu I did a lot of work with substrate and would love to dig into using smoldot with our tooling! Please introduce yourself on our discord and send me a friend request (`@nuke_web3`)
